### PR TITLE
Fix recipes, used by tests

### PR DESCRIPTION
**Pull request by spektrof**
Adjustments were made, based on #1

### CODE REVIEW

File: tests/recipes/patches/pkg_a_add_defines.patch [hunk 1]
```diff
@@ -2,12 +2,8 @@ diff --git a/test2/conan/config/recipes/pkg_a/conanfile.py b/test2/conan/config/
...
--    self.cpp_info.components["pkg_a_lib"].libs = ["lib"]
-\ No newline at end of file
-+    self.cpp_info.components["pkg_a_lib"].libs = ["lib"]
-+    self.cpp_info.components["pkg_a_lib"].defines = ["DEFINE"]
-\ No newline at end of file
+@@ -17,3 +17,4 @@ class pkg_a(ConanFile):
+     self.cpp_info.includedirs = ["include"]
+     self.cpp_info.libdirs = ["lib"]
+     self.cpp_info.libs = ["a"]
++    self.cpp_info.defines = ["DEFINE"]
```
Comment (memsharded) [inline tests/recipes/patches/pkg_a_add_defines.patch]:
```suggestion
     self.cpp_info.libs = ["a"]
     self.cpp_info.defines = ["DEFINE"]
```

Those are the default, can be omitted

File: tests/recipes/pkg_a/conanfile.py [hunk 2]
```diff
@@ -14,7 +14,6 @@ def configure(self):
...
 
   def package_info(self):
-    self.cpp_info.name = "pkg_a"
-    self.cpp_info.components["pkg_a_lib"].name = "pkg_a_lib"
-    self.cpp_info.components["pkg_a_lib"].libdirs = ["lib"]
-    self.cpp_info.components["pkg_a_lib"].libs = ["lib"]
\ No newline at end of file
+    self.cpp_info.includedirs = ["include"]
+    self.cpp_info.libdirs = ["lib"]
+    self.cpp_info.libs = ["a"]
```
Comment (memsharded) [inline tests/recipes/pkg_a/conanfile.py]:
```suggestion
    self.cpp_info.libs = ["a"]
```

File: tests/recipes/pkg_b/conanfile.py [hunk 3]
```diff
@@ -15,7 +15,5 @@ def package_id(self):
     self.info.header_only()
 
   def package_info(self):
-    self.cpp_info.name = "pkg_b"
-    self.cpp_info.components["pkg_b_lib"].name = "pkg_b_lib"
-    self.cpp_info.components["pkg_b_lib"].includes = ["include"]
-    self.cpp_info.components["pkg_b_lib"].requires = ["pkg_a::pkg_a_lib"]
+    self.cpp_info.includedirs = ["include"]
+    self.cpp_info.requires = ["pkg_a::pkg_a"]
```
Comment (memsharded) [inline tests/recipes/pkg_b/conanfile.py]:
```suggestion
    pass
```
requires is no longer necessary, not using components.

File: tests/recipes/pkg_c/conanfile.py [hunk 4]
```diff
@@ -6,6 +6,7 @@ class pkg_c(ConanFile):
   requires = "pkg_b/1.0"
 
   def package_info(self):
-    self.cpp_info.name = "pkg_c"
-    self.cpp_info.components["pkg_c_lib"].name = "pkg_c_lib"
-    self.cpp_info.components["pkg_c_lib"].requires = ["pkg_b::pkg_b_lib"]
+    self.cpp_info.includedirs = ["include"]
+    self.cpp_info.libdirs = ["lib"]
+    self.cpp_info.libs = ["c"]
+    self.cpp_info.requires = ["pkg_b::pkg_b"]
```
Comment (memsharded) [inline tests/recipes/pkg_c/conanfile.py]:
```suggestion
    self.cpp_info.libs = ["c"]
```

File: tests/recipes/pkg_e/conanfile.py [hunk 5]
```diff
@@ -9,5 +9,6 @@ class pkg_e(ConanFile):
   )
 
   def package_info(self):
-    self.cpp_info.name = "pkg_e"
-    self.cpp_info.components["pkg_e_lib"].name = "pkg_e_lib"
+    self.cpp_info.includedirs = ["include"]
+    self.cpp_info.libdirs = ["lib"]
+    self.cpp_info.libs = ["e"]
```
Comment (memsharded) [inline tests/recipes/pkg_e/conanfile.py]:
```suggestion
    self.cpp_info.libs = ["e"]
```

**Comment by spektrof** [comment]
There was a little issue in the changes, now it's good.
Squashed the commits and merge now.

Thank you for checking!

**Pull request merged**